persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java

```diff
@@ -80,11 +82,10 @@ public static BuildProject getBuildProject(Path projectPath) throws BalException
             errorMessage.append("error occurred when validating the project. ");
             int validErrors = 0;
             for (Diagnostic diagnostic : diagnosticResult.errors()) {
+                String[] path = diagnostic.location().lineRange().filePath().split(FILE_PATH_SEPARATOR);
+                String fileName = path[path.length - 1].trim();
 
-                if (!diagnostic.diagnosticInfo().code().equals(DiagnosticErrorCode
-                        .INCOMPATIBLE_TYPES.diagnosticId()) &&
-                        !diagnostic.diagnosticInfo().code().equals(DiagnosticErrorCode
-                        .INCOMPATIBLE_MAPPING_CONSTRUCTOR.diagnosticId())) {
+                if (!Files.exists(projectPath.toAbsolutePath().resolve(GENERATED_SOURCE_DIRECTORY).resolve(fileName))) {
```

Comment:
This file may exist inside the module directory inside the generated directory.

Reply:
Updated in [5f71463](https://github.com/ballerina-platform/persist-tools/pull/83/commits/5f714630b23b5693dfafd59c950713a508753f05)